src/parser.js

```diff
@@ -685,21 +685,28 @@ export default class Parser extends Stream {
               }
               if (dateRange.duration && dateRange.endDate) {
                 const startDate = dateRange.startDate;
-                const newDateInSeconds = startDate.setSeconds(startDate.getSeconds() + dateRange.duration);
+                const newDateInSeconds = startDate.getTime() + (dateRange.duration * 1000);
```

Comment:
Don't know why this was `startDate.setSeconds(startDate.getSeconds() + dateRange.duration);`
I think the `setSeconds` was a typo and was meant to be `getSeconds()'.

For more context this block of code, the spec mentions
`If a Date Range contains both a DURATION attribute and an END-DATE
   attribute, the value of the END-DATE attribute MUST be equal to the
   value of the START-DATE attribute plus the value of the DURATION
   attribute.`
   https://datatracker.ietf.org/doc/html/draft-pantos-http-live-streaming-23#section-4.3.2.7